src/textual/widgets/_data_table.py

```diff
@@ -950,7 +955,63 @@ def watch_cursor_coordinate(
             elif self.cursor_type == "column":
                 self.refresh_column(old_coordinate.column)
                 self._highlight_column(new_coordinate.column)
-            self._scroll_cursor_into_view()
+            # If the coordinate was changed via `move_cursor`, give priority to its
+            # scrolling because it may be animated.
+            self.call_next(self._scroll_cursor_into_view)
+
+    def move_cursor(
```

Comment:
Not sure about the API here personally, because the parameters are kind of dependent on each other. (e.g. if you pass `coordinate` and `row`, things get weird). Do we need the `coordinate` parameter?

Reply:
If you pass a coordinate and a row, you get an error.

I opted for this signature because if the cursor is a coordinate (`DataTable.cursor_coordinate`) it makes sense to accept the same type; then I added `row` and `column` as convenience parameters, which work similarly to the parameters `after` and `before` in methods like `Widget.mount` and `Widget.move_child`

Reply:
I concur with Darren here. The signature can be ambiguous. Given there can be row and column cursors, where only a single value makes sense, I think we should drop the `coordinate`.

Reply:
Even when the cursor type is row/column, the reactive attribute `cursor_coordinate` is a full coordinate with X _and_ Y coordinates, so this is aligned with that.
But if you all think it is a bad idea 🤷